Assets/General/Scripts/Console/ConsoleController.cs

```diff
@@ -227,9 +238,55 @@ private void EnterGodMode()
     private void ExitGodMode() 
     {
         FindObjectOfType<PlayerHealth>().CanPlayerTakeDamage = true;
-        _toggleGodModeButton.GetComponentInChildren<TMP_Text>().text = "Enter God Mode";
+        _toggleGodModeButton.GetComponentInChildren<TMP_Text>().text = "Enter God Mode";   
     }
-    
+
+    #endregion
+
+
+    #region Infinite Focus
+
+    /// <summary>
+    /// puts you in and out of infinite focus mode
+    /// </summary>
+    private void ToggleInfiniteFocus()
+    {     
+        if (_isInfocusState)
+        {
+            ExitInfiniteFocus();
+            _isInfocusState = false;
+
+        }
+        else
+        {
+            EnterInfiniteFocus();
+            _isInfocusState = true;
+        }
+    }
+
+    /// <summary>
+    /// Puts you into infinite focus mode
+    /// </summary>
+    private void EnterInfiniteFocus()
+    {
+        GameObject.FindObjectOfType<HarpoonGun>().CallFocusMax();
+        _infiniteFocusToggleButton.GetComponentInChildren<TMP_Text>().text = "Exit Infinite Focus";
+
+        PlayerManager.Instance.GetOnHarpoonFiredEvent().AddListener(_harpoonGun.CallFocusMax);
+    }
+
+    /// <summary>
+    /// takes you out of infinite focus mode
+    /// </summary>
+    private void ExitInfiniteFocus()
+    {
+        GameObject.FindObjectOfType<HarpoonGun>().CallResetFocus();
+        _infiniteFocusToggleButton.GetComponentInChildren<TMP_Text>().text = "Enter Infinite Focus";
+
+        PlayerManager.Instance.GetOnHarpoonFiredEvent().RemoveListener(_harpoonGun.CallFocusMax);
+    }
+
+
```

Comment:
Noticing some empty space throughout these changes. Not mandatory but would prefer it fixed